src/mpid/ch4/netmod/ofi/ofi_init.h

```diff
@@ -624,23 +629,39 @@ static inline int MPIDI_NM_mpi_init_hook(int rank,
         sprintf(keyS, "OFI-%d", rank);
         MPIDI_OFI_PMI_CALL_POP(PMI_KVS_Put(MPIDI_Global.kvsname, keyS, val), pmi);
         MPIDI_OFI_PMI_CALL_POP(PMI_KVS_Commit(MPIDI_Global.kvsname), pmi);
-        MPIDI_OFI_PMI_CALL_POP(PMI_Barrier(), pmi);
+
+        for (i = 0; i < size; i++) {
+            if (MPIDI_CH4_rank_is_local(i, MPIR_Process.comm_world)) {
+                if (i == rank)
+                    local_rank = num_local;
+
+                if (local_rank_0 == -1)
+                    local_rank_0 = i;
+
+                num_local++;
+            }
```

Comment:
You can't break out of the loop because we also want to calculate the final total of `num_local`, right?